ADALiOS/ADALiOS/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -335,7 +354,7 @@ - (void) validatedAcquireTokenByRefreshToken:(NSString*)refreshToken
           additionalHeaders:nil
                  completion:^(NSDictionary *response)
      {
-         ADTokenCacheStoreItem* resultItem = (cacheItem) ? cacheItem : [ADTokenCacheStoreItem new];
+         ADTokenCacheStoreItem* resultItem = (cacheItem) ? [ADTokenCacheStoreItem initWithStoreItem:cacheItem] : [ADTokenCacheStoreItem new];
```

Comment:
As a general design goal in ADAL we're trying to avoid asynchronously calling backwards in the flow to steps we've already passed, because as you've discovered it's infinite recursive loop bait. In this case we'd probably be better off having it only retry the code it really needs to, which is probably that requestWithServer: call, not the whole validatedAcquireTokenByRefreshToken: call.